### PR TITLE
Prefetch event schedule data to improve initial load speed

### PR DESCRIPTION
**Pull request by joshuayoes**
# Description

[Trello Card](145-prefetch-event-schedule-on-app-load)

One of the benefits of React Query is that we can have patterns like pre-populating a query cache so it seems like a transition is "instant".

This PR separates out `queryKeys` and `queryFns` from hooks so that we can prefetch the event schedule queries on the initial welcome screen. That way, when the user clicks to the event schedule, it feels instantaneous because the data has already been fetched.

This PR also updates Prettier because the existing version had an issue parsing the `satisifies` TypeScript keyword. `yarn format` was also run which updated a few unrelated screens.

# Graphics

The webflow network request in "Before" occur only once the user navigates to the event schedule screen. Whereas the network requests in "After" occur as soon as the app loads.

| Before            | After            |
| -------------- | ------------------ |
|  <video src="https://user-images.githubusercontent.com/37849890/231343209-2b2fd006-6f36-4ad5-aa36-8993de125ad4.mov" /> | <video src="https://user-images.githubusercontent.com/37849890/231343149-6a079591-9d64-4357-8194-8f44721f5ff6.mov" /> |

# Checklist:

- [x] I have done a thorough self-review of my code
- [x] I have tested with a screen reader and font-scaling turned on and added necessary accessibility features
- [x] I have run tests and linter

### CODE REVIEW

File: app/screens/ExploreScreen/ExploreScreen.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface ExploreMapProps {
 }
 
 const recommendationTypes = Object.values(WEBFLOW_MAP.recommendationType)
-type RecommendationType = typeof recommendationTypes[number]
+type RecommendationType = (typeof recommendationTypes)[number]
```
Comment (joshuayoes) [inline app/screens/ExploreScreen/ExploreScreen.tsx:38]: Unrelated change, `yarn format` updated this after updating prettier

File: app/screens/InfoScreen/OurSponsorsScreen.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { colors, spacing } from "../../theme"
 import { SponsorCard } from "./SponsorCard"
 
 const sponsorTiers = Object.values(WEBFLOW_MAP.sponsorTier)
-type Tiers = typeof sponsorTiers[number]
+type Tiers = (typeof sponsorTiers)[number]
```
Comment (joshuayoes) [inline app/screens/InfoScreen/OurSponsorsScreen.tsx:21]: Unrelated change, `yarn format` updated this after updating prettier

File: app/screens/WelcomeScreen.tsx [hunk 3]
```diff
@@ -35,6 +36,10 @@ export const WelcomeScreen: React.FC<WelcomeScreenProps> = (_props) => {
     requestUserPermission()
   }, [])
 
+  useLayoutEffect(() => {
+    prefetchScheduledEvents()
+  }, [])
```
Comment (joshuayoes) [inline app/screens/WelcomeScreen.tsx:41]: Mirrors docs example: https://tanstack.com/query/v4/docs/react/guides/prefetching

File: app/services/api/webflow-api.ts [hunk 4]
```diff
@@ -31,75 +31,137 @@ import {
...
-  useQuery({
-    queryKey: [key],
-    queryFn: async () => {
-      const { data } = await axiosInstance.get<PaginatedItems<T>>(
-        `/collections/${collectionId}/items`,
-      )
-      return data.items
-    },
-    enabled,
-  })
```
Comment (joshuayoes) [inline app/services/api/webflow-api.ts:45]: Decided to replace this with having the `useQuery` options directly accessible, so that we can compose them later with hooks like `useQueries` or prefetching specific queries using their `queryKey` and `queryFn`.

File: app/services/api/webflow-api.ts [hunk 5]
```diff
@@ -31,75 +31,137 @@ import {
...
+const getCollectionById = async <T>(collectionId: string) => {
+  const { data } = await axiosInstance.get<PaginatedItems<T>>(`/collections/${collectionId}/items`)
+  return data.items
+}
+
+const useRecommendationsOptions = {
+  queryKey: [RECOMMENDATIONS.key, RECOMMENDATIONS.collectionId],
```
Comment (joshuayoes) [inline app/services/api/webflow-api.ts:42]: Not sure entirely how necessary this is, but it seems that the docs recommend adding queryKeys for any ids that are associated with a queryFn: https://tanstack.com/query/v4/docs/react/guides/query-keys#if-your-query-function-depends-on-a-variable-include-it-in-your-query-key

I think this is so that we can invalidate specific keys, like if the collectionId ever needed to change for some reason.

File: app/services/api/webflow-api.ts (hunk 4 above)
Comment (mazenchami) [reply app/services/api/webflow-api.ts:45]: it looks like they're the same functions just renamed differently. maybe we can still use this reusable component but expose more of the items you're looking for in the function?

Comment (joshuayoes) [reply app/services/api/webflow-api.ts:45]: The main difference here is that we can directly pass the `queryKey` and `queryFn` instead of accessing it through the `useWebflowAPI` wrapper.

This isn't a huge difference, except exposing the `queryKey` and `queryFn` reference lets use them later in other places like `useQueries` for prefetching multiple queries. Which we can't do if they are wrapped in `useWebflowAPI`.

I thought about exposing `queryKey` and `queryFn` as a part of the existing hooks, but the problem with that is `useQuery` will attempt to fetch using the `queryFn` on mount, which is not ideal for controlling prefetching. That being said,[ I did find a doc example for `enabled`,](https://tanstack.com/query/v4/docs/react/reference/useQuery) which may let us still use the hook but not fetch on mount.
Screenshot 2023-04-12 at 10 30 56 AM

Comment (joshuayoes) [reply app/services/api/webflow-api.ts:45]: Here is the usage later that I'm referring to: https://github.com/infinitered/ChainReactApp2023/pull/108/files#diff-f8eeb82e0e0624e99d187ff5e43f122a24c4a517c0102ad6048e434098d6a843R121-R192

File: app/services/api/webflow-api.ts [hunk 6]
```diff
@@ -31,76 +31,127 @@ import {
...
+  queryFn: async () => getCollectionById<RawRecommendations>(RECOMMENDATIONS.collectionId),
+} satisfies UseQueryOptions
```
Comment (mazenchami) [inline app/services/api/webflow-api.ts:44]: [q] since you're recreating this everywhere, wouldn't it be easier to make a recurring one?

for example:

```tsx
const useWebflowAPI = <T>(key: string, collectionId: string) =>
  useQuery({
    queryKey: [key, collectionId],
    queryFn: async () => {
      const { data } = await axiosInstance.get<PaginatedItems<T>>. 
        (`/collections/${collectionId}/items`)
      return data.items
    }
  })
```

**Review by mazenchami — COMMENTED** [review]
this is a really cool refactor and i think would help us potentially solve #153?

File: app/services/api/webflow-api.ts (hunk 6 above)
Comment (joshuayoes) [reply app/services/api/webflow-api.ts:44]: I'm pretty sure we need the `queryFn` reference to be same when we use it later in `useQueries` in `useScheduledEvents` and `prefetchQuery` in `prefetchScheduledEvents`. But I can check to see if that is true

Comment (joshuayoes) [reply app/services/api/webflow-api.ts]: Refactored this to be less verbose in this commit: https://github.com/infinitered/ChainReactApp2023/pull/108/commits/1127ca1640d4b9f11fe6638fadb6b714e4d2799c

**Review by mazenchami — APPROVED** [review]
LGTM! 🎉 🚀
